documentation/prplMesh-test-boardfarm.md

```diff
@@ -0,0 +1,85 @@
+# Introduction
+
+Current testing scheme is partially virtual.
+All test are done in so-called "dummy" mode.
+This mode operates without any interaction with WiFi device.
+Recently, the number of cases, when issue caused by changes wasn't detected by regular testing (using `test_flows.py`) but noticed during nightly testbed test, has increased.
+Yet, we don't have any means to provide automated testing on real hardware.
+So temporarily, all developers were forced to manually trigger testbed testing for each PR published as part of regular flow.
+CI has only one testbed set up available, so we can perform only single test at same moment.
+Also, testbed is able to perform only Cerification tests.
+We are interested in wider test scenarios, which cover project-specific bugs, corner cases and etc.
+
+Due to limitations of current testing approach, need of automated tests on real hardware raises.
+In order to speed up development process, it was decided by community to use [boardfarm](https://github.com/mattsm/boardfarm/) as a base for automated testing.
+This project can provide us with basic infrastructure for tests and save effort in terms of development and support.
+For details about integration process, design please see next sections.
+
+# Design
+
+By itself, [boardfarm](https://github.com/mattsm/boardfarm/) provides us with:
+1. Abstraction of test set-up
+1. Provides some basic classes of devices
+1. Some simple tests
+1. Handles connections to boards
+1. Provides test-suite organisation
+
+[boardfarm](https://github.com/mattsm/boardfarm/) already has some representation of some general devices (like Linux, OpenWRT device).
+But our test flows involve more components and require more info about devices\[1\].
+So we have to expand base classes with representation of `prplWRT` devices capable handling Agent, Controller and Sniffer.
+Also, some additional abstraction of easyMesh specific entities may ease future work(like CAPI, IEEE1905.1 message).
+
+Hierarchy of devices, entities their relation can be understood easier using typical test set-up scheme provided below:
+
+![test setup](/documentation/images/plantuml/boardfarm_prplMesh_test_setup.png)
+
+Please, note.
+In general case, it's expected that prplMesh executables are delivered to DUT (device under test) as OpenWRT `*.ipk`.
+DUT should be already burn with prplWRT or other supported platform.
+Thus, LAN connection required for flashing in uboot mode is missing from the scheme.
+
+In general, in order to start using [boardfarm](https://github.com/mattsm/boardfarm/) we have to create:
+
+1. Devices
+	1. prplWRT
+	1. dockerized dummy device
+	1. Wired sniffer
+	1. Wireless sniffer
+1. Additional entities
+	1. CAPI
+	1. prplMesh_controller
+	1. prplMesh_agent
+	1. prplMesh_radio_dummy
+	1. prplMesh_radio_hw
```

Comment:
It would be better to use the actual names that we end up using:
```suggestion
	1. UCCSocket
	1. ALEntity
        1. Radio
            1. RadioDocker
            1. RadioHostapd
        1. VirtualAP
            1. VirtualAPDocker
            1. VirtualAPHostapd
```

Bonus points for updating the UML files with these names as well :-)

Reply:
I Have added a new commit with UML update. Let me know if got it right.